src/types/tabs.ts

```diff
@@ -1,4 +1,7 @@
+import type { RouteLocationNamedRaw } from 'vue-router'
+
 export interface Tab {
   hash: string
   title: string
+  route?: RouteLocationNamedRaw
```

Comment:
**Comment**: I generally use the named variant of `RouteLocationRaw` for such interfaces even though it limits the route styles you can use (i.e. disallowing path-based routes) because it makes working with the objects implementing `Tab` easier. They will always be able to access `tab.route.name` which is not possible with `RouteLocationRaw` (as it can be just a string or miss the `name` property from an object). I think this is fine because it’s best to keep routes addressed via path out of application code (i.e. paths should only be used in route definitions) as that allows for easier changes to an application’s routes (one only has to change them in the route definitions, all uses of `router.push`, etc. remain intact).